judge/comments.py

```diff
@@ -78,10 +79,14 @@ def post(self, request, *args, **kwargs):
             try:
                 parent = int(parent)
             except ValueError:
+                return HttpResponseBadRequest()
+            try:
+                parent_comment = Comment.objects.get(hidden=False, id=parent, page=page)
+            except Comment.DoesNotExist:
                 return HttpResponseNotFound()
-            else:
-                if not Comment.objects.filter(hidden=False, id=parent, page=page).exists():
-                    return HttpResponseNotFound()
+            if not (self.request.user.has_perm('judge.change_comment') or
+                    parent_comment.time > timezone.now() - settings.DMOJ_COMMENT_REPLY_TIMEFRAME):
```

Comment:
I might be the only one, but I feel like

```
if timezone.now() - parent_comment.time > settings.DMOJ_COMMENT_REPLY_TIMEFRAME and not self.request.user.has_perm('judge.change_comment'):
```

Is the more readable way of writing this.